### PR TITLE
Add support for non-looping complete screen animations

### PR DESCRIPTION
**Pull request by Kalobi**
Currently, any animation on complete screens except for the parallax must loop indefinitely. This PR implements support for non-looping layers by adding a new meta attribute Loop to complete screen layers, true by default, that causes the animation to stay on the last frame if set to false. This is implemented both for .meta.yaml and .xml so that it can be used both for custom maps and global skinmods.

### CODE REVIEW

File: Celeste.Mod.mm/Patches/CompleteRenderer.cs [hunk 1]
```diff
@@ -97,7 +98,47 @@ public UILayerNoXML(CompleteRenderer renderer, MapMetaCompleteScreenLayer meta)
 
         }
 
-        public class ImageLayerNoXML : ImageLayer {
+        public class patch_ImageLayer : ImageLayer {
+
+            public bool Loop;
+            public patch_ImageLayer(Vector2 offset, Atlas atlas, XmlElement xml) : base(offset, atlas, xml) {
+                //no-op
+            }
```
Comment (coloursofnoise) [inline Celeste.Mod.mm/Patches/CompleteRenderer.cs]:
```suggestion
            public patch_ImageLayer(Vector2 offset, Atlas atlas, XmlElement xml)
                : base(offset, atlas, xml) {
                //no-op
            }
```

File: Celeste.Mod.mm/MonoModRules.cs [hunk 2]
```diff
@@ -2473,6 +2479,17 @@ bgt.un.s	ldloca.s
             cursor.RemoveRange(3);
         }
 
+        public static void PatchCompleteRendererImageLayerRender(ILContext context, CustomAttribute attrib) {
+            MethodReference m_ImageLayer_Render = MonoModRule.Modder.FindType("Celeste.CompleteRenderer/ImageLayer")
+                .Resolve().FindProperty("ImageIndex").GetMethod;
```
Comment (coloursofnoise) [inline Celeste.Mod.mm/MonoModRules.cs]:
```suggestion
            MethodReference m_ImageLayer_Render = context.Method.DeclaringType.FindProperty("ImageIndex").GetMethod;
```

File: Celeste.Mod.mm/MonoModRules.cs [hunk 3]
```diff
@@ -2473,6 +2479,17 @@ bgt.un.s	ldloca.s
...
+            
+            ILCursor cursor = new(context);
```
Comment (coloursofnoise) [inline Celeste.Mod.mm/MonoModRules.cs]:
```suggestion
            ILCursor cursor = new ILCursor(context);
```

**Review by coloursofnoise — COMMENTED** [review]
Some style nitpicks, fields should all be defined at the top of the class, with an empty line before the constructor.

Some of these should probably be added to the `.editorconfig`, such as target-typed new expressions :sweat_smile:

File: Celeste.Mod.mm/MonoModRules.cs (hunk 3 above)
Comment (Kalobi) [reply Celeste.Mod.mm/MonoModRules.cs]: What's the issue with target-typed new expressions?

Comment (Kalobi) [reply Celeste.Mod.mm/MonoModRules.cs]: Fixed

**Review by coloursofnoise — APPROVED** [review]
LGTM 👍

**Pull request merged**